### ISSUE TITLE
Cloudtrail mode with "mark-for-op" action with 0 days not working as expected

### ISSUE THREAD

**Issue by mkundrapu** [capitalone/cloud-custodian#2489]
Team,
   we wrote a policy to add custom tags while creating dynamodb table and we used “cloudtrail” mode with actions “mark-for-op” with days as 0. Our understand is that, if we provide 0 as value, the value of the date in the tag value will be the same day when the event is triggered. But this policy executed the value of the date in tag  is 2018/06/11, but we expect it be 2018/06/07. Can you please look in to it? 

Note: This is the same issue we observed with other resource types too.

Policy:
    mode:
       type: cloudtrail
       tags:
         *custodian-tags
       events:
         - event: CreateTable
           source: "dynamodb.amazonaws.com"
           ids: "requestParameters.tableName"
    filters:
       - or: *required-tags-absent
    actions:
       - type: mark-for-op
         tag: custodian_tagging
         msg: "This DynamoDB table is missing at least one of the required tags: {op}@{action_date}"
         days: 0
         op: delete

This is the tag value generated:
        This DynamoDB table is missing at least one of the required tags: delete@2018/06/11

**Comment by thisisshi** [comment]
do you have another policy that will actually perform the delete? you need a `mark-for-op` policy and a corresponding policy that actually does the action

**Comment by mkundrapu** [comment]
yes, we do have the corresponding policy. But it's not deleting the resource as the value is not today, so it might delete it on 2018/06/11 as per the example above.

resource: dynamodb-table
    filters:
      - type: marked-for-op
        tag: custodian_tagging
        op: delete
      - or: *required-tags-absent
    actions:
      - delete

**Comment by thisisshi** [comment]
Ah sorry, misread the original question

**Comment by kapilt** [comment]
what if you tried `days: 0.1`

**Comment by mkundrapu** [comment]
Let me try with 0.1 as value and get back to you.
Agree to your suggestion for Dynamodb table. 
But we want to use this logic for different resource types too. So if the value 0.1 works then we are good, if not would like you to look in to it further. 

thanks @kapilt and @thisisshi

**Comment by mkundrapu** [comment]
@kapilt , it's not allowing 0.1 due to integer validation. Please let me know if you want me to try anything else.

**Comment by thisisshi** [comment]
I was actually able to re-create this behavior, I have a branch where I realized that our dynamodb test regarding mark-for-op is actually bad, i'll look into why exactly it's adding 4 days ahead.

Branch: https://github.com/thisisshi/cloud-custodian/tree/dynamodb-mark-test-fix

**Comment by thisisshi** [comment]
Actually... it's fairly obvious why this is behaving the way that it is: we're defaulting to 4 days if days == 0: https://github.com/capitalone/cloud-custodian/blob/master/c7n/tags.py#L610

Not exactly sure why this is the default behavior, couldn't find anything in the discussion from the PR that it came from, this should be 0 imo as this default behavior is seemingly random i.e. why 4 and not 5? or 6? or 7? if we want to block 0 as a input, that should happen at the validation stage, picking a date for the user seems very dangerous.

**Comment by thisisshi** [comment]
@mkundrapu here's the PR for the fix, thanks for the bug report https://github.com/capitalone/cloud-custodian/pull/2498

**Comment by mkundrapu** [comment]
@thisisshi thanks for the update. Is this fix is only for dynamodb? I see the same behavior in few resource types too. I can provide the details if required.

**Comment by thisisshi** [comment]
this will fix for all resources, dynamodb just had some additional test coverage that I wanted to add

**Comment by mkundrapu** [comment]
@thisisshi , great thank you!!!
